### PR TITLE
Transparent background for themed icons

### LINKED ISSUE #3003: [FEATURE]  Transparent icon background
### Describe the feature

It would be nice to have an option to make icon background transparent when themed icons (with Lawnicons) used. Below is a workaround that I'm using wallpaper with same color as icon background.

### Why would it be useful to add?

That could bring visual appeal to the home screen by eliminating some clutter.

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

Screenshot_20221005-192327

Screenshot_20221005-192337

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by naveenccmsd**
## Description

It would be nice to have an option to make icon background transparent when themed icons (with Lawnicons) used.

Closes #3003

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
 :white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

Impementation step.

- Added switch to enable `Transparent Background` as Experimental option.
- If `Transparent Background` Enabled background would be set to transparent.
     - No other change applied to dark mode
     - Previous Icon color will be odd or to transparent in light theme without icon background. So composite color will be calculared for current Icon color and same will be applied.

### CODE REVIEW

**Comment by naveenccmsd** [comment]
[3003#issuecomment-1273121884](https://github.com/LawnchairLauncher/lawnchair/issues/3003#issuecomment-1273121884)

**Comment by kahanzammi** [comment]
Tested build 1181 on Android 13. Lawnchair freezes as soon as themed icons enabled - without and without enabling the experiment feature. Unable to capture any logs. Screenshot below.

Screenshot_20221011-171757

**Comment by naveenccmsd** [comment]
Even I'm facing same issue from build 1181 but didn't face such issue from latest changes ,

**Comment by naveenccmsd** [comment]
Please wait for latest build to approved and executed

**Comment by kahanzammi** [comment]
Tested latest, works fine. I believe currently only black icons are supported.

Screenshot_20221011-223332

**Comment by naveenccmsd** [comment]
> Tested latest, works fine. I believe currently only black icons are supported.

No,  Try chnaging system as dark mode.

In Light theme mode icon color is grey/black  and background is accent color whereas in dark mode icon color is accent color and background is greay/black

Please find sample.

photo_2022-10-11_02-04-55
photo_2022-10-11_02-04-55 (2)

**Comment by dhirenvasnani** [comment]
Works fine, but idk why clock's icon still has a background 
IMG_20221012_005713
@naveenccmsd

**Comment by naveenccmsd** [comment]
Because it's dynamic icon from lawnchair override, let me see if this can fixed.

**Comment by yasanglass** [comment]
Thank you for contributing to Lawnchair. If the PR is not ready for review for now, please convert it to a draft PR as it's currently a bit confusing if we are supposed to review it yet or not.

**Comment by naveenccmsd** [comment]
@yasandev , Sorry , I didn't noticed the clock icon didn't removed the bg as I don't  have it . It's fixed now.
Please review.

**Comment by naveenccmsd** [comment]
Screenshot_20221012-025443_Lawnchair

This will be like this after change

**Review by Goooler — CHANGES_REQUESTED** [review]
Changes in platform_frameworks_libs_systemui need to be split to a new PR in https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui

**Comment by naveenccmsd** [comment]
@Goooler created new PR 

https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui/pull/7

**Comment by Goooler** [comment]
@naveenccmsd Please update submodule to https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui/commit/b8244916420ff8e49382d75637e31353212c24a7.

**Comment by yasanglass** [comment]
Not sure how you included the actual submodule file changes on this PR, but the submodule change on this PR should only be the target commit being changed as @Goooler just noted.

**Review by Goooler — APPROVED** [review]

File: lawnchair/res/values/strings.xml [hunk 1]
```diff
@@ -57,6 +57,8 @@
     <string name="workspace_increase_max_grid_size_description">Increase the max allowed Home Screen grid size from 10 x 10 to 20 x 20.</string>
     <string name="always_reload_icons_label">Always Reload Icons</string>
     <string name="always_reload_icons_description">Avoid using cached icons from icon packs</string>
+    <string name="transparent_background_icons">Transparent Background</string>
+    <string name="transparent_background_icons_description">Icon color might updated to match with background in Light Theme</string>
```
Comment (yasanglass) [inline lawnchair/res/values/strings.xml]:
```suggestion
    <string name="transparent_background_icons">Transparent Themed Icons</string>
    <string name="transparent_background_icons_description">Use transparent background on themed icons</string>
```

These strings can be more clear.

**Comment by Goooler** [comment]
Synced, wait for other reviewers.

**Review by yasanglass — CHANGES_REQUESTED** [review]

**Comment by yasanglass** [comment]
Also, it would probably be better if the preference included "ThemedIcons" in its name as it would avoid confusion in the future.

**Comment by naveenccmsd** [comment]
> Synced, wait for other reviewers.

Did I messed up  ? I still see submodule  changes in the PR.

**Comment by Goooler** [comment]
Nope, the git parent repo just points to a commit in the submodule.

**Review by yasanglass — APPROVED** [review]

**Pull request merged**